### ISSUE TITLE
Traffic sign detection, too many false positives

### ISSUE THREAD

**Issue by majkaz** [mapillary/mapillary_issues#2565]
The problem is quite extreme here and combined with missing editor, it makes the traffic sign recognition unusable. 

[Example](https://www.mapillary.com/app/?lat=48.989048430000025&lng=14.475994140000012&z=17&signs=true&ocr=true&pKey=DWotPIp3kXNr-LCSyabMcA&mapStyle=osm&focus=photo) - 4 traffic signs recognized, only one is correct and 3 are wrong. 

Some of the problems could be avoided by recognizing the road geometry - then it is clear where one can expect the real traffic signs (not on [cars](https://www.mapillary.com/app/?lat=48.98867261999996&lng=14.477694779999979&z=17&signs=true&ocr=true&pKey=umwReNRGrUCLe-sveGimdw&mapStyle=osm&focus=photol), for example, and probably ignoring the [overhead signs with directions](https://www.mapillary.com/app/?lat=48.98776335000002&lng=14.4810248&z=17&signs=true&ocr=true&pKey=dxNO91qxUYAgqHTpMJDQog&mapStyle=osm&focus=photo) as well and on the edges of the photo - see the right edge [here](https://www.mapillary.com/app/?lat=48.988511500000016&lng=14.478293880000024&z=17&signs=true&ocr=true&pKey=YPIJCowAlDxQOlpZbnf5Gw&mapStyle=osm&focus=photo)) and by fine tuning the comparison - the OCR image usually shows something quite a bit different from the "recognized" sign.

As a workaround at the moment, is there a way to get the "old" Mapillary layer without the recognition in JOSM?

Or can anybody say when it will be possible to edit the traffic signs?

**Comment by gyllen** [comment]
Hi @majkaz 

A while ago we actually had a traffic sign editor but for some performance reasons we had to pull it back. You would be happy to hear though that we currently are working on something new and better. Keep on following our blog and hopefully we will deliver something that will work for you. As for the core traffic sign detector we are constantly working on improving it. By having an edit toll we will be able to use this edit tool to train even better.

For JOSM Im not 100% sure wich layer you mean but I think @peterneubauer would be the right person to answer that question.

**Comment by peterneubauer** [comment]
I think for JOSM, given that both the detections and Objects already are fetched, there might be interest to edit these are push the data back to the APIs to make recognition better. cc @floscher

**Comment by floscher** [comment]
@majkaz I believe you are referring to the traffic signs with the orange boxes around them that are painted on top of the image display in JOSM. These are on by default, but you can easily turn them off:
* open the image info panel using the ![mapillary-info svg](https://cloud.githubusercontent.com/assets/3904348/26169603/ac01bb98-3b3e-11e7-981a-015cee019202.png) icon in the left sidebar of JOSM
* uncheck the checkbox in the image info panel reading "show detections on top of image"
* this setting is sticky, so it will stay the same when you'll view different images/sequences or when you restart JOSM


@peterneubauer Is there already an API endpoint for sending audits on image detections? Last time we talked about it (https://github.com/JOSM/Mapillary/issues/46#issuecomment-293543172) it was still in the works and I can't find anything about it in the [API docs](https://www.mapillary.com/developer/api-documentation/)

**Comment by yopaseopor** [comment]
The problem is to miss the possibility to edit via JOSM or web the mapillary wrong detection and turn back thant information to Mapillary. Also it would be interesting to merge the same traffic sign, detected in 2 or three sequences but there are the same traffic sign in the reality. Also it would be interesting to "stick some tracks in the same street and direction " to an existent way in OSM. Then , with all this information together the detection would be more accurate due to the merging of the duplicities. Keep pushing Mapillary!

**Comment by peterneubauer** [comment]
That is true. The APIs for feedback and correction of detections are coming soon, so they will be able to integrate with JOSM.
